Added test manager documentation.

Jeffrey-Simpson opened the pull request:
Added test manager information to the read the docs documentation.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gridappsd/gridappsd-docs/7)
<!-- Reviewable:end -->

Jeffrey-Simpson committed:
Added test manager

craigpnnl reviewed (commented):
Please add a preamble to tell us exactly what the test manager allows us to do....as you can see from the screen shot of the built docs just going into the configuration seems like an abrupt transition.  
![image](https://user-images.githubusercontent.com/3979063/40861764-1eb941fc-659f-11e8-8d83-03189dc21d14.png)
Note if this is elsewhere in the docs please reference this here.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+
+1. Test Configuration
+The test configuration contains information to define the simulation run,
```
Add a space between Test Configuration and "The test" so that a hard return will show up in the built docs.  I am in the process of having documentation on how to build the docs through a docker image at present so that will be easier to test soon.

Jeffrey-Simpson replied:
I will add a line.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+  }
+
+2. Test Script
+The test script contains the name of the test script and the name of the
```
Same comment adding a line space to make it go to the second line.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+
+
+3. Expected results series:
+Time series json structure with expected results.
```
Same line in between

Jeffrey-Simpson replied:
I will add a line.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+2. Test Script
+The test script contains the name of the test script and the name of the
+application, it needs to match the name of application on platform.
+It also contains the test configuration path, the outputs to to listen for,
```
Add a link to how to mount an application in the platform.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+  {
+      "expected_outputs": {
+          "1": {
+              "simulation_id": "2034749692",
```
How will the simulation_id be determined?  As far as I know each run will get a different value.

Jeffrey-Simpson replied:
I will add words describing this. That value is a placeholder that must be replaced with the simulation id.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+              "simulation_id": "2034749692",
+              "message": {
+                  "timestamp": "2018-05-03 18:24:20.464399",
+                  "measurements": [
```
Above you have a start and end date in 2017, but this message is sent in 2018...seems like a disconnect here.

Jeffrey-Simpson replied:
Good catch. I will ifx.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+  }
+
+
+4. Rules
```
Step 4 seems like background information...is this part of the expected results above and the rules are similar?  What am I as an app developer supposed to do at this step?

Jeffrey-Simpson replied:
Removed background information and added an example.

craig8 commented on docs/source/using_gridappsd/test_manager.rst:
```diff
@@ -0,0 +1,134 @@
+
+
+
+Run request_test.py provided for the sample app
```
Is this done inside the container?  if so please change user@foo to what it will be when running from gridappsd-docker/.run.sh script.

What directory is the request_test.py in?  I believe that unless the user has mounted the sample app  this file isn't in the sample app and hte user doesn't run it from there this will not work...let me know if I am incorrect.

Jeffrey-Simpson replied:
I have been running from outside the container, mostly in PyCharm. I will update this step to clarify

Jeffrey-Simpson committed:
Fixes based on comments.
Added overview.
Spacing for RST format.
Better rules example.
Added clarity for the request_test script.

Jeffrey-Simpson committed:
Merge branch 'master' of https://github.com/GRIDAPPSD/gridappsd-docs

Jeffrey-Simpson committed:
Added Solar Forecasting app

Jeffrey-Simpson committed:
Intial checkin of test_manager_events.rst

Jeffrey-Simpson committed:
Updated code blocks, and query

Jeffrey-Simpson committed:
Merge branch 'master' into master

Jeffrey-Simpson committed:
Updated diagram

Jeffrey-Simpson committed:
Updated for new command structure.

Jeffrey-Simpson committed:
Merged

Jeffrey-Simpson committed:
Removed titles I has put in for some reason

Jeffrey-Simpson commented:
I have updated and resolved the comments but I still see a change requested. Is there something else I need to do?

Jeffrey-Simpson committed:
Updated CIM Fault Event

Jeffrey-Simpson committed:
Updated Comm and Fault events

Jeffrey-Simpson committed:
Updated

Jeffrey-Simpson committed:
Updated

Jeffrey-Simpson committed:
Merged with master

Jeffrey-Simpson committed:
Merged

AAndersn commented:
@poorva1209 I suggest closing this old ticket

This has already been covered in new docs:

https://gridappsd-training.readthedocs.io/en/latest/api_usage/3.5-Creating-Running-Simulation-API.html#Test-Manager-Configuration

https://gridappsd-training.readthedocs.io/en/latest/installation/1.5-Using-GridAPPS-D-Viz.html#Test-Manager-Configuration
